classes/Deck.js

```diff
@@ -1,22 +1,115 @@
-import { PlayingCard } from './PlayingCard'
+import { PlayingCard, SUITES, RANKS } from './playing-card'
 
 class Deck {
-  constructor(suites, ranks) {
+  constructor(cards) {
+    this.cards = cards
+  }
+
+  static from(suites, ranks) {
     const cards = []
+    let i = 0
     for (const suite of suites) {
       for (const rank of ranks) {
-        cards.push(new PlayingCard(suite, rank))
+        cards.push(new PlayingCard(suite, rank, i++))
       }
     }
-    this.cards = cards
+    return cards
+  }
```

Comment:
I don't understand the reason for this method. The name is misleading (I would think an instance of a Deck is returned from it), and it unnecessarily exposes a piece of its implementation (the use of PlayingCards) that is unnecessary to be known. Like, in any case you are passing in those two arrays, so why pass it to this method just for it to return some array of newly constructed objects and then pass it to the constructor?

I think you should just decide that the user must pass in two arrays of `suites` and `ranks` to the constructor, and create the array of cards there like you had in the previous pull request.